File: review-agent-service/app/models/review.py
"""审核领域与 API 模型。

主要职责：定义请求、响应、证据、比较和任务状态契约。
修改日期：2026-08-26
修改人：wuyi
"""

from enum import StrEnum
from typing import Any, Literal

from pydantic import (
    AliasChoices,
    BaseModel,
    ConfigDict,
    Field,
    HttpUrl,
    model_validator,
)

from app.agent.models import (
    AgentAdvice,
    CheckResult,
    MaterialCompletenessReport,
    RetrySummary,
)
from app.models.checks import CheckResultValue
from app.models.evidence import (  # noqa: F401 - public export
    EvidenceFact,
    FieldObservation,
)


class FieldStatus(StrEnum):
    MATCH = "MATCH"
    CONFLICT = "CONFLICT"
    REVIEW_REQUIRED = "REVIEW_REQUIRED"


class Recommendation(StrEnum):
    PASS = "PASS"
    REJECT_SUGGESTED = "REJECT_SUGGESTED"
    REVIEW_REQUIRED = "REVIEW_REQUIRED"


class JobStatus(StrEnum):
    RUNNING = "RUNNING"
    PARTIAL = "PARTIAL"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"


class BusinessType(StrEnum):
    SCRAP_REPLACEMENT = "scrap_replacement"
    VEHICLE_SOURCE = "vehicle_source"
    TRANSFER = "transfer"  # deprecated; no active profile
    CONSISTENCY = "consistency"


class Region(StrEnum):
    DEFAULT = "default"
    QINGDAO = "qingdao"
    CHANGCHUN = "changchun"


class SelectionMode(StrEnum):
    AUTO = "AUTO"
    MANUAL = "MANUAL"


class ReviewDisplayTarget(StrEnum):
    PAGE_FIELD = "PAGE_FIELD"
    ASSISTANT = "ASSISTANT"


class ImageInput(BaseModel):
    model_config = ConfigDict(extra="ignore", populate_by_name=True)

    index: int = Field(ge=0)
    src: str
    group: str = "未分类"
    image_id: str | None = Field(
        default=None, validation_alias=AliasChoices("image_id", "imageId")
    )
    category_hint: str = Field(
        default="unknown",
        validation_alias=AliasChoices("category_hint", "categoryHint"),
    )
    business_scope: str = Field(
        default="unknown",
        validation_alias=AliasChoices("business_scope", "businessScope"),
    )
    group_title: str = Field(
        default="未分类资料",
        validation_alias=AliasChoices("group_title", "groupTitle"),
    )
    group_order: int | None = Field(
        default=None,
        ge=1,
        validation_alias=AliasChoices("group_order", "groupOrder"),
    )
    document_type_hint: str = Field(
        default="unknown",
        validation_alias=AliasChoices("document_type_hint", "documentTypeHint"),
    )
    mime_type: str | None = Field(
        default=None, validation_alias=AliasChoices("mime_type", "mimeType")
    )
    size_bytes: int | None = Field(
        default=None, ge=0, validation_alias=AliasChoices("size_bytes", "sizeBytes")
    )
    data_url: str | None = Field(
        default=None, validation_alias=AliasChoices("data_url", "dataUrl")
    )
    collection_error: str | None = Field(
        default=None,
        validation_alias=AliasChoices("collection_error", "collectionError"),
    )
    alt: str = ""
    page_position: str | None = Field(
        default=None,
        validation_alias=AliasChoices("page_position", "pagePosition"),
    )

    @model_validator(mode="before")
    @classmethod
    def derive_legacy_business_scope(cls, value: Any) -> Any:
        if (
            not isinstance(value, dict)
            or "business_scope" in value
            or "businessScope" in value
        ):
            return value
        category = value.get("category_hint", value.get("categoryHint", "unknown"))
        old_types = {"scrap_certificate", "old_vehicle", "registration_certificate"}
        new_types = {"new_vehicle", "invoice"}
        scope = (
            "old_vehicle"
            if category in old_types
            else "new_vehicle"
            if category in new_types
            else "unknown"
        )
        return {**value, "business_scope": scope}




class FieldComparison(BaseModel):
    field: str
    left_value: Any = None
    right_value: Any = None
    status: FieldStatus
    source: str = "rule_engine"
    confidence: float | None = Field(default=None, ge=0, le=1)
    evidence: list[EvidenceFact] = Field(default_factory=list)
    message: str = ""
    differences: list[int] = Field(default_factory=list)


class QrCheck(BaseModel):
    image_index: int | None = None
    raw_value: str | None = None
    url: str | None = None
    domain_valid: bool | None = None
    accessible: bool | None = None
    page_fields: dict[str, Any] = Field(default_factory=dict)
    status: FieldStatus = FieldStatus.REVIEW_REQUIRED
    message: str = ""


class PageFillAction(BaseModel):
    field: str
    target_label: str
    owner_type: str


class CapabilityResult(BaseModel):
    capability_id: str
    status: Literal["READY", "SKIPPED", "BLOCKED", "NOT_CONFIGURED", "SUCCEEDED", "FAILED"]
    checks: list[CheckResult] = Field(default_factory=list)
    evidence: list[EvidenceFact] = Field(default_factory=list)
    limitations: list[str] = Field(default_factory=list)
    page_actions: list[PageFillAction] = Field(default_factory=list)
    qr_checks: list[QrCheck] = Field(default_factory=list)
    display_items: list[str] = Field(default_factory=list)
    error_code: str | None = None
    attempts: int = 0


class CapabilityPlanEntry(BaseModel):
    """Serializable execution plan entry exposed by ReviewResponse."""

    capability_id: str
    status: Literal["READY", "SKIPPED", "BLOCKED", "NOT_CONFIGURED"]
    stage: str
    reason: str = ""
    dependencies: list[str] = Field(default_factory=list)
    missing_dependencies: list[str] = Field(default_factory=list)


class ReviewTask(BaseModel):
    """一个已经实际执行、可按顺序展示的审核步骤。"""

    step_id: str
    check_ids: list[str] = Field(default_factory=list)
    sequence: int = Field(ge=1)
    category: Literal["FIELD", "EXTERNAL", "BUSINESS_RULE", "MATERIAL"]
    display_target: ReviewDisplayTarget
    page_field: str | None = None
    page_target_field: str | None = None
    page_value: Any = None
    control_type: str | None = None
    writable: bool = False
    requires_reviewer_action: bool
    label: str
    result_status: Literal["MATCH", "CONFLICT", "INSUFFICIENT"]
    reason: str
    values: list[CheckResultValue] = Field(default_factory=list)
    evidence: list[EvidenceFact] = Field(default_factory=list)
    details: dict[str, Any] = Field(default_factory=dict)
    evidence_count: int = 0
    evidence_mode: str | None = None
    evidence_sources: list[str] = Field(default_factory=list)
    not_found_sources: list[str] = Field(default_factory=list)
    normalized_page_value: str | None = None

    @model_validator(mode="after")
    def validate_display_contract(self) -> "ReviewTask":
        if self.display_target is ReviewDisplayTarget.PAGE_FIELD and not self.page_field:
            raise ValueError("PAGE_FIELD review steps require page_field")
        # Assistant tasks use page_target_field for controlled DOM location and
        # write-back. ``page_field`` remains reserved for PAGE_FIELD display.
        if self.display_target is ReviewDisplayTarget.ASSISTANT and self.page_field:
            raise ValueError("ASSISTANT review steps must not bind page_field; use page_target_field")
        if self.result_status == "MATCH" and self.requires_reviewer_action:
            raise ValueError("MATCH review steps cannot require reviewer action")
        if self.result_status != "MATCH" and not self.requires_reviewer_action:
            raise ValueError("non-MATCH review steps must require reviewer action")
        return self


# Backward-compatible import for clients that still use the pre-v2 name.
# The object is the same validated contract; new code should use ReviewTask.
ReviewStep = ReviewTask



class ReviewFieldSnapshot(BaseModel):
    """One editable logical form control collected from the current page DOM."""

    model_config = ConfigDict(extra="ignore", populate_by_name=True)

    field: str | None = None
    label: str
    value: Any = None
    control_type: str = Field(
        default="text", validation_alias=AliasChoices("control_type", "controlType")
    )
    editable: bool = True
    order: int = Field(default=1, ge=1)
    section: str = "unknown"
    operation_only: bool = Field(
        default=False,
        validation_alias=AliasChoices("operation_only", "operationOnly"),
    )


class ReviewRequest(BaseModel):
    model_config = ConfigDict(extra="ignore", populate_by_name=True)

    page_url: HttpUrl | str = Field(
        validation_alias=AliasChoices("page_url", "pageUrl")
    )
    business_type: BusinessType = Field(
        default=BusinessType.SCRAP_REPLACEMENT,
        validation_alias=AliasChoices("business_type", "businessType"),
    )
    region: Region = Region.DEFAULT
    profile_version: str = Field(
        default="1.0",
        validation_alias=AliasChoices("profile_version", "profileVersion"),
    )
    selection_mode: SelectionMode = Field(
        default=SelectionMode.AUTO,
        validation_alias=AliasChoices("selection_mode", "selectionMode"),
    )
    workflow_stage: str = Field(
        default="scrap_replacement",
        validation_alias=AliasChoices("workflow_stage", "workflowStage"),
    )
    application_id: str | None = Field(
        default=None,
        validation_alias=AliasChoices("application_id", "applicationId"),
    )
    page_title: str = Field(
        default="",
        validation_alias=AliasChoices("page_title", "pageTitle"),
    )
    page_fields: dict[str, Any] = Field(
        default_factory=dict,
        validation_alias=AliasChoices("page_fields", "pageFields"),
    )
    review_fields: list[ReviewFieldSnapshot] = Field(
        default_factory=list,
        validation_alias=AliasChoices("review_fields", "reviewFields"),
    )
    images: list[ImageInput] = Field(default_factory=list)
    collection_diagnostics: "CollectionDiagnostics" = Field(
        default_factory=lambda: CollectionDiagnostics(),
        validation_alias=AliasChoices(
            "collection_diagnostics", "collectionDiagnostics"
        ),
    )
    trace_id: str | None = Field(default=None, validation_alias=AliasChoices("trace_id", "traceId"))


class CollectionDiagnostics(BaseModel):
    model_config = ConfigDict(extra="ignore", populate_by_name=True)

    scanned_controls: int = Field(
        default=0,
        ge=0,
        validation_alias=AliasChoices("scanned_controls", "scannedControls"),
    )
    matched_fields: int = Field(
        default=0,
        ge=0,
        validation_alias=AliasChoices("matched_fields", "matchedFields"),
    )
    review_field_count: int = Field(
        default=0,
        ge=0,
        validation_alias=AliasChoices("review_field_count", "reviewFieldCount"),
    )
    unmatched_labels: list[str] = Field(
        default_factory=list,
        validation_alias=AliasChoices("unmatched_labels", "unmatchedLabels"),
    )
    candidate_count: int = Field(
        default=0,
        ge=0,
        validation_alias=AliasChoices("candidate_count", "candidateCount"),
    )
    ambiguous_fields: list[str] = Field(
        default_factory=list,
        validation_alias=AliasChoices("ambiguous_fields", "ambiguousFields"),
    )
    image_success_count: int = Field(
        default=0,
        ge=0,
        validation_alias=AliasChoices("image_success_count", "imageSuccessCount"),
    )
    image_failure_count: int = Field(
        default=0,
        ge=0,
        validation_alias=AliasChoices("image_failure_count", "imageFailureCount"),
    )
    scanned_images: int = Field(
        default=0,
        ge=0,
        validation_alias=AliasChoices("scanned_images", "scannedImages"),
    )
    selected_images: int = Field(
        default=0,
        ge=0,
        validation_alias=AliasChoices("selected_images", "selectedImages"),
    )
    image_overflow: bool = Field(
        default=False, validation_alias=AliasChoices("image_overflow", "imageOverflow")
    )
    collection_issues: list[str] = Field(
        default_factory=list,
        validation_alias=AliasChoices("collection_issues", "collectionIssues"),
    )


class ResultSection(BaseModel):
    id: str
    title: str
    fields: list[str] = Field(default_factory=list)


class PageActionIntent(BaseModel):
    """Backend-proposed reversible page action; execution stays in the adapter."""

    action_id: str
    payload: dict[str, Any] = Field(default_factory=dict)
    requires_authorization: bool = True


class ReviewResponse(BaseModel):
    capability_plan: list[CapabilityPlanEntry] = Field(default_factory=list)
    capability_results: list[CapabilityResult] = Field(default_factory=list)
    evidence_facts: list[EvidenceFact] = Field(default_factory=list)
    protocol_version: Literal["2.0"] = "2.0"
    trace_id: str = ""
    presentation: Literal["FIELD_WORKBENCH", "MANUAL_REVIEW"] = "MANUAL_REVIEW"
    business_type: BusinessType = BusinessType.SCRAP_REPLACEMENT
    region: Region = Region.DEFAULT
    profile_version: str = "1.0"
    recommendation: Recommendation
    risk_level: str
    summary: str
    comparisons: list[FieldComparison] = Field(default_factory=list)
    qr_checks: list[QrCheck] = Field(default_factory=list)
    cross_checks: list[CheckResult] = Field(default_factory=list)
    issues: list[str] = Field(default_factory=list)
    sections: list[ResultSection] = Field(default_factory=list)
    context_summary: dict[str, Any] = Field(default_factory=dict)
    agent_advice: AgentAdvice | None = None
    material_completeness: MaterialCompletenessReport | None = None
    retry_summary: RetrySummary | None = None
    page_fill_intent: list[PageFillAction] = Field(default_factory=list)
    page_actions: list[PageActionIntent] = Field(default_factory=list)
    review_tasks: list[ReviewTask] = Field(default_factory=list)


class ReviewProgress(BaseModel):
    total_count: int = 0
    completed_count: int = 0
    failed_count: int = 0
    timed_out_count: int = 0


class ReviewGroupProgress(ReviewProgress):
    status: JobStatus = JobStatus.RUNNING


class ReviewJobCreated(BaseModel):
    job_id: str
    status: JobStatus = JobStatus.RUNNING
    created_at: str


class ReviewJobSnapshot(BaseModel):
    job_id: str
    status: JobStatus
    created_at: str
    progress: ReviewProgress
    groups: dict[str, ReviewGroupProgress] = Field(default_factory=dict)
    result: ReviewResponse | None = None
    message: str | None = None
    material_completeness: MaterialCompletenessReport | None = None
    retry_summary: RetrySummary | None = None

File: review-extension/src/types/review.ts
export interface DifferenceRange { kind: "REPLACE" | "EXTRA" | "MISSING"; start: number; end: number; page_start: number; page_end: number; page_text: string; }
export type ImageGroup = "身份证" | "回收证明" | "旧车资料" | "新车资料" | "发票及其他图片" | "未分类";
export type BusinessType = "scrap_replacement" | "vehicle_source" | "consistency";
export type Region = "default" | "qingdao" | "changchun";
export type SelectionMode = "AUTO" | "MANUAL";

export interface BusinessSelection {
  businessType: BusinessType;
  region: Region;
  profileVersion: string;
  workflowStage: BusinessType;
  selectionMode: SelectionMode;
}

/** Content Script 从审核页提取的图片元数据；不包含图片二进制。 */
export interface PageImage {
  index: number;
  imageId?: string;
  src: string;
  alt: string;
  group: ImageGroup | string;
  categoryHint?: string;
  documentTypeHint?: string;
  businessScope?: string;
  groupTitle?: string;
  groupOrder?: number;
  mimeType?: string | null;
  sizeBytes?: number | null;
  dataUrl?: string | null;
  collectionError?: string | null;
  pagePosition?: string;
  naturalWidth?: number;
  naturalHeight?: number;
}

export interface WritableTargetSnapshot {
  field: "old_vehicle.affiliation" | "new_vehicle.affiliation";
  label: string;
  present: boolean;
  currentValue: string | null;
}

/** 页面字段 DOM 目标的跨消息快照；真实元素只保留在 Content Script 内部。 */
export interface PageFieldTargetSnapshot {
  field: string;
  present: boolean;
}

/** 当前页面实际存在的可操作表单控件，顺序和数量以本次 DOM 采集为准。 */
export interface ReviewFieldSnapshot {
  field?: string | null;
  label: string;
  value: string;
  controlType: string;
  editable: boolean;
  order: number;
  section: string;
  operationOnly?: boolean;
}

export interface PageData {
  pageUrl: string;
  sourceTabId: number;
  pageInstanceId: string;
  pageFingerprint: string;
  collectionId: string;
  pageTitle: string;
  applicationId?: string;
  pageFields: Record<string, string>;
  reviewFields?: ReviewFieldSnapshot[];
  writableTargets?: WritableTargetSnapshot[];
  fieldTargets?: PageFieldTargetSnapshot[];
  pageText: string;
  images: PageImage[];
  businessType?: BusinessType;
  region?: Region;
  profileVersion?: string;
  workflowStage?: BusinessType;
  selectionMode?: SelectionMode;
  businessDetectionError?: string;
  /** 该次采集已被更新的采集取代；面板应提示重新采集而不是继续提交。 */
  staleCollection?: boolean;
  collectionDiagnostics?: {
    scannedControls: number;
    matchedFields: number;
    reviewFieldCount?: number;
    unmatchedLabels: string[];
    candidateCount?: number;
    ambiguousFields?: string[];
    imageSuccessCount: number;
    imageFailureCount: number;
    scannedImages?: number;
    selectedImages?: number;
    imageOverflow?: boolean;
    collectionIssues?: string[];
  };
  collectionIssues?: string[];
}

export interface MaterialCompletenessIssue {
  code: string;
  field_details?: Array<{
    field: string;
    field_label: string;
    material_name: string;
    value: unknown;
    image_id?: string | null;
    image_index?: number | null;
  }>;
  reason_code?: "material_missing" | "image_unreadable" | "recognition_failed" | "recognition_uncertain" | "evidence_not_extracted" | "page_field_missing" | null;
  reason_detail?: string | null;
  material_type?: string | null;
  business_scope?: string | null;
  field?: string | null;
  required_pages?: number[];
  present_pages?: number[];
  missing_pages?: number[];
  missing_sources?: string[];
  message: string;
  suggested_action: string;
}

export interface MaterialChecklistItem {
  key: string;
  display_name: string;
  material_type: string;
  business_scope: string;
  status: "PRESENT" | "MISSING" | "UNCERTAIN";
  required_pages: number[];
  present_pages: number[];
  missing_pages: number[];
  image_ids: string[];
  reason: string;
}

export interface MaterialCompletenessReport {
  phase: "COLLECTED" | "EXTRACTED";
  status: "COMPLETE" | "INCOMPLETE" | "UNCERTAIN";
  enforced: boolean;
  issues: MaterialCompletenessIssue[];
  checklist?: MaterialChecklistItem[];
}

export interface RetryAttempt {
  target_id: string;
  stage: "qwen" | "qr_decode" | "qr_web";
  attempt_number: number;
  reason_code: string;
  strategy: string;
  result: "succeeded" | "failed" | "skipped";
  duration_ms: number;
}

export interface RetrySummary {
  total_retries?: number;
  qwen_retries?: number;
  qr_decode_rounds?: number;
  qr_web_retries?: number;
  attempts: RetryAttempt[];
}

export type FieldStatus = "MATCH" | "CONFLICT" | "REVIEW_REQUIRED";
export type Recommendation = "PASS" | "REJECT_SUGGESTED" | "REVIEW_REQUIRED";
export type CheckResultStatus = "MATCH" | "CONFLICT" | "INSUFFICIENT";

export interface CheckResultValue {
  differences?: DifferenceRange[];
  source: string;
  value?: unknown;
  source_id?: string | null;
  image_id?: string | null;
  image_index?: number | null;
  document_type?: string | null;
  detail?: string | null;
  derived_from?: string | null;
  evidence_region?: number[] | null;
}

export interface CheckResult {
  check_id: string;
  label: string;
  status: CheckResultStatus;
  reason: string;
  values: CheckResultValue[];
  evidence?: EvidenceFact[];
  details?: ReviewTaskDetails;
}

export interface SubjectEvidenceRequirement {
  party: "OLD_VEHICLE" | "NEW_VEHICLE" | "SHARED";
  subject_name: string;
  subject_type: "PERSONAL" | "COMPANY";
  document: "identity_card_front" | "identity_card_back" | "business_license";
  status: "PRESENT" | "MISSING" | "UNCERTAIN";
  image_ids: string[];
  reason: string;
}

export interface ReviewTaskDetails {
  differences?: number[];
  subject_requirements?: SubjectEvidenceRequirement[];
  auxiliary_checks?: AuxiliaryCheckSummary[];
  affiliation_subject_status?: CheckResultStatus;
}

export interface AuxiliaryCheckSummary {
  check_id: string;
  label: string;
  status: CheckResultStatus;
  reason: string;
}

export interface EvidenceFact {
  source: string;
  uncertain?: boolean;
  field?: string | null;
  source_id?: string | null;
  image_index?: number;
  detail?: string;
  image_id?: string | null;
  business_scope?: string | null;
  group_title?: string | null;
  group_order?: number | null;
  document_type?: string | null;
  value?: unknown;
  normalized_value?: string | null;
  derived_from?: string | null;
  evidence_region?: number[] | null;
  conflicting?: boolean;
}

declare global {
  var ReviewEvidenceLabel: {
    label(evidence: EvidenceFact): string;
    documentNames: Record<string, string>;
  };
}

export interface FieldComparison {
  field: string;
  left_value?: string | null;
  right_value?: string | null;
  status: FieldStatus;
  source: string;
  confidence?: number | null;
  evidence: EvidenceFact[];
  message: string;
  differences?: number[];
}

export interface QrCheck {
  image_index?: number | null;
  raw_value?: string | null;
  url?: string | null;
  domain_valid?: boolean | null;
  accessible?: boolean | null;
  page_fields: Record<string, string>;
  status: FieldStatus;
  message: string;
}

export interface ReviewResponse {
  protocol_version: "2.0";
  presentation: "FIELD_WORKBENCH" | "MANUAL_REVIEW";
  business_type: BusinessType;
  region: Region;
  profile_version: string;
  recommendation: Recommendation;
  risk_level: string;
  summary: string;
  comparisons: FieldComparison[];
  qr_checks: QrCheck[];
  cross_checks: CheckResult[];
  issues: string[];
  sections: ResultSection[];
  context_summary?: {
    field_count?: number;
    image_count?: number;
    image_failed_count?: number;
    completed_count?: number;
    failed_count?: number;
    timed_out_count?: number;
  };
  agent_advice?: {
    decision: "PASS" | "REVIEW_REQUIRED";
    title: string;
    summary: string;
    confidence?: number | null;
    recognition_confidence?: number | null;
    findings: CheckResult[];
    basis: string[];
    limitations: string[];
  } | null;
  material_completeness?: MaterialCompletenessReport | null;
  retry_summary?: RetrySummary | null;
  capability_plan?: CapabilityPlanEntry[];
  capability_results?: CapabilityResult[];
  page_fill_intent?: PageFillAction[];
  page_actions?: PageActionIntent[];
  review_tasks?: ReviewTask[];
}

export interface PageActionIntent {
  action_id: string;
  payload: Record<string, unknown>;
  requires_authorization: boolean;
}

export interface CapabilityPlanEntry {
  capability_id: string;
  status: "READY" | "SKIPPED" | "BLOCKED" | "NOT_CONFIGURED";
  stage: string;
  reason: string;
  dependencies: string[];
  missing_dependencies: string[];
}

export interface CapabilityResult {
  capability_id: string;
  status: "READY" | "SKIPPED" | "BLOCKED" | "NOT_CONFIGURED" | "SUCCEEDED" | "FAILED";
  checks?: CheckResult[];
  evidence?: EvidenceFact[];
  qr_checks?: QrCheck[];
  display_items?: string[];
  limitations?: string[];
  error_code?: string | null;
  attempts?: number;
}

export type ReviewDisplayTarget = "PAGE_FIELD" | "ASSISTANT";

export interface ReviewTask {
  step_id: string;
  sequence: number;
  category: "FIELD" | "EXTERNAL" | "BUSINESS_RULE" | "MATERIAL";
  display_target: ReviewDisplayTarget;
  page_field?: string | null;
  /** Canonical DOM field for sidebar tasks; distinct from PAGE_FIELD rendering. */
  page_target_field?: string | null;
  page_value?: unknown;
  control_type?: string | null;
  writable?: boolean;
  requires_reviewer_action: boolean;
  label: string;
  result_status: CheckResultStatus;
  reason: string;
  values: CheckResultValue[];
  evidence: EvidenceFact[];
  details?: ReviewTaskDetails;
  evidence_count?: number;
  evidence_mode?: string | null;
  evidence_sources?: string[];
  not_found_sources?: string[];
  normalized_page_value?: string | null;
}

export interface PageFillAction {
  field: "old_vehicle.affiliation" | "new_vehicle.affiliation";
  target_label: string;
  owner_type: "PERSONAL" | "COMPANY";
}

/** 审核员从助手中选择候选值后发起的单字段页面回填。 */
export interface PageWriteAction {
  field: string;
  value: string;
  expectedValue?: string | null;
}

export interface ResultSection {
  id: string;
  title: string;
  fields: string[];
}

export type JobStatus = "RUNNING" | "PARTIAL" | "COMPLETED" | "FAILED";

export interface ReviewProgress {
  total_count: number;
  completed_count: number;
  failed_count: number;
  timed_out_count: number;
}

export interface ReviewGroupProgress extends ReviewProgress {
  status: JobStatus;
}

export interface ReviewJobCreated {
  job_id: string;
  status: JobStatus;
  created_at: string;
}

export interface ReviewJobSnapshot {
  job_id: string;
  status: JobStatus;
  created_at: string;
  progress: ReviewProgress;
  groups: Record<string, ReviewGroupProgress>;
  result?: ReviewResponse | null;
  message?: string | null;
  material_completeness?: MaterialCompletenessReport | null;
  retry_summary?: RetrySummary | null;
}
